v2/pkg/genruntime/admissions.go

```diff
@@ -50,10 +51,16 @@ func ValidateWriteOnceProperties(oldObj ARMMetaObject, newObj ARMMetaObject) (ad
 	ownerAdded := oldOwner == nil && newOwner != nil
 	ownerRemoved := oldOwner != nil && newOwner == nil
 
-	if (bothHaveOwner && oldOwner.Name != newOwner.Name) || ownerAdded {
-		errs = append(errs, errors.Errorf("updating 'Owner.Name' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldObj.GetName()))
+	ownerNameChanged := bothHaveOwner && oldOwner.Name != newOwner.Name
+	ownerARMIDChanged := bothHaveOwner && oldOwner.ARMID != newOwner.ARMID
+	if ownerAdded {
+		errs = append(errs, errors.Errorf("adding an owner to an already created resource is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldObj.GetName()))
+	} else if ownerNameChanged {
+		errs = append(errs, errors.Errorf("updating 'spec.owner.name' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldObj.GetName()))
+	} else if ownerARMIDChanged {
+		errs = append(errs, errors.Errorf("updating 'spec.owner.armId' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldObj.GetName()))
 	} else if ownerRemoved {
-		errs = append(errs, errors.Errorf("removing 'Owner' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldObj.GetName()))
+		errs = append(errs, errors.Errorf("removing 'spec.owner' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldObj.GetName()))
 	}
```

Comment:
There are four faillure cases here, but I only see three related tests - do we need to add a test for `adding an owner ...`?

Reply:
I'm actually not sure it's possible to hit that error in practice, because it requires an Azure resource to support existing both without an owner, but also with an owner.

I can't think of any resource type that's actually true for. I think that check is just us trying to be defensive. I've added a comment to that effect on the first clause here.